media/js/firefox/firstrun/learnmore/yahoo-search.js

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function(Mozilla, $) {
+    'use strict';
+
+    var client = Mozilla.Client;
+    var $tryButton = $('#try-it');
+    var $pointer = $('.pointer');
+
+    /**
+     * Check the user has search input in their pallet and Yahoo! is
+     * in their list of possible search providers.
+     * @returns Promise (resolve) if both requirements are met.
+     */
+    function shouldPromoteYahooSearch() {
+
+        var searchProviders = new Promise(function(resolve, reject) {
+            Mozilla.UITour.getConfiguration('search', function(config) {
+                if (config && config.engines) {
+                    resolve(config.engines);
+                } else {
+                    reject('UITour: search engines array not found.');
+                }
+            });
+        });
+
+        var availableTargets = new Promise(function(resolve, reject) {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+                if (config && config.targets) {
+                    resolve(config.targets);
+                } else {
+                    reject('UITour: targets property not found.');
+                }
+            });
+        });
+
+        return new Promise(function(resolve, reject) {
+            Promise.all([searchProviders, availableTargets]).then(function(results) {
+                var engines = results[0];
+                var targets = results[1];
+                resolve(targets && targets.indexOf('search') > -1 && engines && engines.indexOf('searchEngine-yahoo'));
```

Comment:
Do we need a `> -1` check on `engines.indexOf('searchEngine-yahoo')`? If `searchEngine-yahoo` happened to be in position 0, could we get unexpected results here?